app.py

```diff
@@ -35,91 +36,105 @@
-                # If a single file was downloaded; fallback to mp4 (which probably isn't the best idea)
-                merged_ext = result.get('ext', 'mp4')
 
-        # Return the sanitized file path with for processing
-        video_file = os.path.join(app.config['UPLOAD_FOLDER'], sanitized_title + "." + merged_ext)
-        return os.path.abspath(video_file)
+class Utils:
+    """Utility class for common operations like file cleanup and sanitization."""
+
+    @staticmethod
+    def cleanup_old_files(base_filename):
+        """Removes any existing files with the same base name."""
+        file_paths = [
+            os.path.join(app.config['UPLOAD_FOLDER'], base_filename + '.webm'),
+            os.path.join(app.config['UPLOAD_FOLDER'], base_filename + '_isolated_audio.wav'),
+            os.path.join(app.config['UPLOAD_FOLDER'], base_filename + '_processed_video.mp4')
+        ]
+        for path in file_paths:
+            if os.path.exists(path):
+                logging.info(f"Removing old file: {path}")
+                os.remove(path)
+
+    @staticmethod
+    def sanitize_filename(filename):
+        """Replace non-alphanumerics (except periods and underscores) with underscores."""
+        return re.sub(r'[^a-zA-Z0-9._-]', '_', filename)
+
+
+class VideoProcessor:
+    """Class to handle video download and processing logic."""
+
+    @staticmethod
+    def download_media(url):
+        try:
+            # Extract media info first to sanitize title
+            with yt_dlp.YoutubeDL() as ydl:
+                info_dict = ydl.extract_info(url, download=False)
+                base_title = info_dict['title']
+                sanitized_title = Utils.sanitize_filename(base_title)
+
+            ydl_opts = {
+                'format': 'bestvideo+bestaudio/best',
+                'outtmpl': os.path.join(app.config['UPLOAD_FOLDER'], sanitized_title + '.%(ext)s'),
+                'noplaylist': True,
+                'keepvideo': True,  # Keep source files for merging
+                'n_threads': 6
+            }
+
+            with yt_dlp.YoutubeDL(ydl_opts) as ydl:
+                result = ydl.extract_info(url, download=True)
+
+                # Post-download file-ext gymnastics
+                if 'requested_formats' in result:
+                    # If separate video and audio were downloaded and merged
+                    merged_ext = result['ext']
+                else:
+                    # If a single file was downloaded; fallback to mp4 (which probably isn't the best idea)
+                    merged_ext = result.get('ext', 'mp4')
+
+            # Return the sanitized file path for processing
+            video_file = os.path.join(app.config['UPLOAD_FOLDER'], sanitized_title + "." + merged_ext)
+            return os.path.abspath(video_file)
+
+        except Exception as e:
+            logging.error(f"Error downloading video: {e}")
+            return None
+
+    @staticmethod
+    def process_media_with_cpp(video_path):
```

Comment:
Let's rename this to `process_with_media_processor`. Since it's already a part of MediaHandler, no reason anymore to be explicit about we're processing media in the method name, but we should be explicit about the MediaProcessor.